exmg-paper-combobox.html

```diff
@@ -0,0 +1,473 @@
+          if (this._inputValue.length > 0 && items[i].textContent.indexOf(this._inputValue) === -1) {
+            items[i].setAttribute('hidden', '');
+          } else {
+            items[i].removeAttribute('hidden');
+          }
+        }
+      }
+
+      _observeSelectedItem(selectedItem) {
+        if (!selectedItem) {
+          this.set('_token', null);
+          return;
+        }
+
+        const id = this.attrForSelected ?
+          selectedItem.getAttribute(this.attrForSelected) :
+          this.indexOf(selectedItem);
+
+        const text = this.selectedItemSelector ?
+          selectedItem.querySelector(this.selectedItemSelector).textContent :
+          selectedItem.textContent;
+
+        this.set('_token', {id, text});
+      }
+
+      _handleDeleteToken(e) {
+        e.stopPropagation();
+        this.set('selectedValue', null);
+
+        this._resetInput();
+      }
+
+      indexOf(item) {
+        return this.$.listbox.items ? this.$.listbox.items.indexOf(item) : -1;
+      }
+
+      _handleClick(e) {
+        const inside = e.path.find((path) => path === this);
+
+        // Detect outside element click for auto validate input
+        if (this.autoValidate && (this.previousInsideClick && !inside) || this._token) {
+          this.validate();
+        }
+        this.previousInsideClick = inside;
+      }
+
+      _handleAddToken(e) {
+        e.stopPropagation();
+        this._resetInput();
+      }
+
+      connectedCallback() {
+        super.connectedCallback();
+        /* Initialize the input helper span element for determining the actual width of the input
+          text. This width will be used to create a dynamic width on the input field */
+        this.$.inputWidthHelper.style = window.getComputedStyle(this.$.inputValue, null).cssText;
+        this.$.inputWidthHelper.style.position = 'absolute';
+        this.$.inputWidthHelper.style.top = '-999px';
+        this.$.inputWidthHelper.style.left = '0';
+        this.$.inputWidthHelper.style.padding = '0';
+        this.$.inputWidthHelper.style.width = 'auto';
+        this.$.inputWidthHelper.style['white-space'] = 'pre';
+
+        this.$.inputValue.addEventListener('keydown', this._boundKeyDown);
+        if (this.autoValidate) {
+          window.addEventListener('click', this._boundOutsideClick);
+        }
+      }
+
+      disconnectedCallback() {
+        super.disconnectedCallback();
+        this.$.inputValue.removeEventListener('keydown', this._boundKeyDown);
+        if (this.autoValidate) {
+          window.removeEventListener('click', this._boundOutsideClick);
+        }
+      }
+
+      _handleKeyDown(e) {
+        this._inputValue = this._inputValue || '';
+        switch (e.keyCode) {
+          case BACKSPACE:
+          case DELETE:
+            this.opened = true;
+            this.set('selectedValue', null);
+            this.focus();
+            break;
+          case ARROWDOWN:
+          case ENTER:
```

Comment:
remove enter -> just add focus to reset method so when item changed is detected through handleAddtoken -> _resetInput the focus will be restored -> at this moment restoring focus is not working in all use cases